### PR TITLE
[Keymap] adding new user (p4yne) layout with complex lighting feature…

### PR DESCRIPTION
**Pull request by p4yne**
…s (per layer, per key, per type) and usefull layers DE/US, etc.

## Description

* Eight layers (DE, LYR_SW_MEDIA, MOUSE_DSKTP_NAV, DYN_MACRO_NUMPAD_LEADER, GAMING, US, US_SHIFTED, DE_FLAG_LIGHTNING)
* Different color for each layer (DRK_RED, RED, MAGENTA, PURPLE, DRK_GOLD, DRK_BLUE, BLUE, DRK_RED)
* Keys with something else defined than KC_TRNS are lit up in the layer color, so you easily can see which keys have some function defined
* Custom layer types for KC_TRNS keys to let lower layer shine through, or stay black/off
* Custom lighting based on layer, key and led flag 
* Caps-Lock lights up when active

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

NONE

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c
Comment (drashna) [inline keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c:17]: I would recommend renaming the `keymap.h` file to prevent collisions with `quantum/keymap.h`.

File: keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c
Comment (drashna) [inline keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c:19]: Shouldn't be needed. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c
Comment (p4yne) [reply keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c:17]: Most of the keymaps I looked at had the same pattern. keymap.c+keymap.h isn't c-style good practice to have those files share their name?

Comment (p4yne) [reply keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c:17]: @drashna Is the rename blocking the merge? Feedback is much appreciated. Thank you.

**Comment by p4yne** [comment]
Can a maintainer please help me understand what is expected and why I don't get any feedback on my questions? Thank you.

File: keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c
Comment (drashna) [reply keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c:17]: The biggest issue is that `/quantum/keymap.h` exists. And that collision may cause serious issues.

Comment (p4yne) [reply keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c:17]: okay, thank you then I'll rename it this was not clear to me from your initial response.

**Comment by p4yne** [comment]
@drashna Thank you, please review again.

File: keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c
Comment (precondition) [inline keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c:207]: Why go through the process of undefining and redefining `_______` just for a line that's commented out and inactive anyways?

File: keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/custom_keymap.h
Comment (drashna) [inline keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/custom_keymap.h:60]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/custom_keymap.h
Comment (drashna) [inline keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/custom_keymap.h:60]: No longer exists. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/custom_keymap.h
Comment (p4yne) [reply keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/custom_keymap.h]: Thank you. Fixed.

File: keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c
Comment (drashna) [inline keyboards/kprepublic/bm65hsrgb_iso/keymaps/p4yne/keymap.c:223]:
```suggestion
    uprintf("KL: kc: 0x%04X, col: %u, row: %u, pressed: %u, time: %u, interrupt: %u, count: %u\n", keycode, record->event.key.col, record->event.key.row, record->event.pressed, record->event.time, record->tap.interrupted, record->tap.count);
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by keyboard-magpie — APPROVED** [review]

**Pull request merged**
